### PR TITLE
Handle Console Operator LogLevel

### PR DESCRIPTION
**Pull request by jhadvig**
Comments in code

/assign @benjaminapetersen 

Story: https://jira.coreos.com/browse/CONSOLE-1320

### CODE REVIEW

File: pkg/console/subresource/deployment/deployment.go [hunk 1]
```diff
@@ -206,6 +225,7 @@ func consoleContainer(cr *operatorv1.Console) corev1.Container {
 			"--public-dir=/opt/bridge/static",
 			"--config=/var/console-config/console-config.yaml",
 			"--service-ca-file=/var/service-ca/service-ca.crt",
+			flag,
```
Comment (jhadvig) [inline pkg/console/subresource/deployment/deployment.go]: Adding both, flag and argument(above), so the log-level is passed to the `bridge` binary

File: pkg/console/operator/operator.go [hunk 2]
```diff
@@ -250,6 +250,7 @@ func (c *consoleOperator) defaultConsoleOperatorConfig() *operatorsv1.Console {
 		Spec: operatorsv1.ConsoleSpec{
 			OperatorSpec: operatorsv1.OperatorSpec{
 				ManagementState: operatorsv1.Managed,
+				LogLevel:        operatorsv1.Debug,
```
Comment (jhadvig) [inline pkg/console/operator/operator.go]: The log-level is set here

Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: So the `- "-v=4"` there is for the `operator` pod.

We want to pipe logging down into the `console` pod.

Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: So we will need to pipe the log level here:

https://github.com/openshift/console-operator/blob/master/pkg/console/subresource/deployment/deployment.go#L204

Which I imagine will require a bit of a translation function from `operatorsv1.Debug` to whatever `capnslog` wants for `Debug`.  They may be the same, maybe not.  But probably will be some kind of `--log-level=*=DEBUG` string creation.

File: pkg/console/subresource/deployment/deployment.go [hunk 3]
```diff
@@ -196,6 +198,20 @@ func consoleVolumeMounts(vc []volumeConfig) []corev1.VolumeMount {
...
+	case operatorv1.Debug:
+		level = "DEBUG"
+	case operatorv1.Trace:
+		fallthrough
+	case operatorv1.TraceAll:
+		fallthrough
+	default:
+		level = "TRACE"
+	}
+	flag := fmt.Sprintf("--log-level=logrus=%s", level)
```
Comment (benjaminapetersen) [inline pkg/console/subresource/deployment/deployment.go]: `logrus?`

File: pkg/console/operator/operator.go (hunk 2 above)
Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: OK this should work, though should we set it to `NORMAL` rather than `DEBUG`?

Comment (jhadvig) [reply pkg/console/operator/operator.go]: yep, just wanted to see what it will spit out :)

Comment (jhadvig) [reply pkg/console/operator/operator.go]: @benjaminapetersen updated

**Comment by jhadvig** [comment]
@benjaminapetersen updated

File: pkg/console/subresource/deployment/deployment.go [hunk 4]
```diff
@@ -206,6 +224,7 @@ func consoleContainer(cr *operatorv1.Console) corev1.Container {
 			"--public-dir=/opt/bridge/static",
 			"--config=/var/console-config/console-config.yaml",
 			"--service-ca-file=/var/service-ca/service-ca.crt",
+			flag,
```
[the remark this answers is not in the document]
Comment (benjaminapetersen) [reply pkg/console/subresource/deployment/deployment.go]: Ok so to ensure the pod template changes enough that `ApplyDeployment()` can notice, lets do this:

```yaml
  template:
    metadata:
      annotations:
        console.openshift.io/console-config-version: "1622686"
        console.openshift.io/image: docker.io/openshift/origin-console:latest
        console.openshift.io/oauth-secret-version: "10542"
        console.openshift.io/service-ca-config-version: "10545"
        # add a log level annotation to kick it
        console.openshift.io/log-level: TRACE
```
That will ensure that `modified=true` which will cause an update to roll out.

**Comment by benjaminapetersen** [comment]
If we don't want to use the `annotation` that way (we are kind of abusing it) we can do something like:

```golang
// in func SyncDeployment()
deployment, _, err := resourceapply.ApplyDeployment(
        c.deployments,
        c.recorder,
        expectedDeployment,
        resourcemerge.ExpectedDeploymentGeneration(expectedDeployment, operatorConfig.Status.Generations),
        // redeploy on operatorConfig.spec changes
        operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration, 
    )

// so long as somewhere in sync_v400() we add 
operatorConfig.Status.ObservedGeneration = operatorConfig.ObjectMeta.Generation
// at the end of the sync loop before we update status. 
// generation increments on spec changes, so this would probably work better as we dont have to 
// do bookkeeping on the deployment itself to ensure it rolls out, we can just trust that 
// changes to our operator config will flip the boolean
```

That seems better.

File: pkg/console/operator/operator.go (hunk 2 above)
Comment (benjaminapetersen) [reply pkg/console/operator/operator.go]: Sorry meant `Normal` should be the default right?
Even though the `console` `deployment` itself is really sparse on logs...

**Comment by benjaminapetersen** [comment]
/retest

`failed to apply using Terraform`

**Comment by benjaminapetersen** [comment]
lgtm so long as tests pass.

**Comment by jhadvig** [comment]
```
fail [github.com/openshift/origin/test/extended/deployments/deployments.go:399]: Expected
    <string>: --> pre: Running hook pod ...
    --> pre: Success
    --> Scaling up deployment-test-2 from 0 to 1, scaling down deployment-test-1 from 0 to 0 (keep 1 pods available, don't exceed 2 pods)
        Scaling deployment-test-2 up to 1
    --> Success
to contain substring
    <string>: test pre hook executed
```

/retest

**Pull request merged**